### PR TITLE
Document when MemoryStream.TryGetBuffer returns false

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes: https://github.com/dotnet/dotnet-api-docs/issues/3604

Source code: https://source.dot.net/#System.Private.CoreLib/MemoryStream.cs

CC @svick

### CODE REVIEW

**Comment by carlossanlop** [comment]
Thanks for the approval, @Thraka .
There were warnings in the xrefs. I'm investigating how to indicate constructor DocIDs, because the `#` does not seem to be right.

File: xml/System.IO/MemoryStream.xml [hunk 1]
```diff
@@ -2016,11 +2016,29 @@ The pending operation does not support writing.</exception>
...
+          <format type="text/markdown">
+  <![CDATA[
+
+## Remarks
+
+This method returns `true` when the the underlying buffer is marked as exposable, which happens when the current `MemoryStream` instance is created with:
+
+- <xref:System.IO.MemoryStream.#ctor>.
+- <xref:System.IO.MemoryStream.#ctor(System.Int32)>.
+- <xref:System.IO.MemoryStream.#ctor(System.Byte[],System.Int32,System.Int32,System.Boolean,System.Boolean)> with the parameter `publiclyVisible` set to `true`.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]: you need to encode # 

```suggestion
- <xref:System.IO.MemoryStream.%23ctor(System.Byte[],System.Int32,System.Int32,System.Boolean,System.Boolean)> with the parameter `publiclyVisible` set to `true`.
```

File: xml/System.IO/MemoryStream.xml [hunk 2]
```diff
@@ -2016,11 +2016,29 @@ The pending operation does not support writing.</exception>
...
+          <see langword="true" /> if the buffer is exposable; otherwise, <see langword="false" />.</returns>
+        <remarks>
+          <format type="text/markdown">
+  <![CDATA[
+
+## Remarks
+
+This method returns `true` when the the underlying buffer is marked as exposable, which happens when the current `MemoryStream` instance is created with:
+
+- <xref:System.IO.MemoryStream.#ctor>.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
- <xref:System.IO.MemoryStream.%23ctor>.
```

File: xml/System.IO/MemoryStream.xml [hunk 3]
```diff
@@ -2016,11 +2016,29 @@ The pending operation does not support writing.</exception>
...
+- <xref:System.IO.MemoryStream.#ctor(System.Int32)>.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
- <xref:System.IO.MemoryStream.%23ctor(System.Int32)>.
```

File: xml/System.IO/MemoryStream.xml [hunk 4]
```diff
@@ -2016,11 +2016,29 @@ The pending operation does not support writing.</exception>
...
+- <xref:System.IO.MemoryStream.#ctor(System.Byte[],System.Int32,System.Int32,System.Boolean,System.Boolean)> with the parameter `publiclyVisible` set to `true`.
+
+The underlying buffer will not be exposed if the current `MemoryStream` instance is created with:
+
+- <xref:System.IO.MemoryStream.#ctor(System.Byte[],System.Boolean)>.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
- <xref:System.IO.MemoryStream.%23ctor(System.Byte[],System.Boolean)>.
```

File: xml/System.IO/MemoryStream.xml [hunk 5]
```diff
@@ -2016,11 +2016,29 @@ The pending operation does not support writing.</exception>
...
+- <xref:System.IO.MemoryStream.#ctor(System.Byte[],System.Int32,System.Int32,System.Boolean,System.Boolean)> with the parameter `publiclyVisible` set to `false`.
```
Comment (mairaw) [inline xml/System.IO/MemoryStream.xml]:
```suggestion
- <xref:System.IO.MemoryStream.%23ctor(System.Byte[],System.Int32,System.Int32,System.Boolean,System.Boolean)> with the parameter `publiclyVisible` set to `false`.
```

**Pull request merged**